src/util/piped_process.h

```diff
@@ -0,0 +1,94 @@
+/// \file piped_process.h
+/// Subprocess communication with pipes
+/// \author Diffblue Ltd.
+
+#ifndef CPROVER_UTIL_PIPED_PROCESS_H
+#define CPROVER_UTIL_PIPED_PROCESS_H
+
+#ifdef _WIN32
+// TODO: Windows definitions go here
+#else
+
+#  include <vector>
+
+#  include "optional.h"
+
+#  define PIPED_PROCESS_INFINITE_TIMEOUT                                       \
+    optionalt<std::size_t>                                                     \
+    {                                                                          \
+    }
+
+class piped_processt
```

Comment:
📖 Give an overview (with example) of how this API should be used.

Reply:
It's now under `doc/API/util/piped_process.md`.